dio/lib/src/transformers/util/consolidate_bytes.dart

```diff
@@ -0,0 +1,12 @@
+import 'dart:typed_data';
+
+/// Consolidates a stream of [Uint8List] into a single [Uint8List]
+Future<Uint8List> consolidateBytes(Stream<Uint8List> stream) async {
```

Comment:
isn't this the same as `collectBytes(Stream<List<int> stream)` in the `async` package that dio already uses?

Reply:
They don't share the same arguments, no?

Reply:
cc @knaeckeKami

Reply:
An Uint8List implements List<int>, so I guess it would work. 
The implementation also looks very similar. 


```dart

Future<Uint8List> collectBytes(Stream<List<int>> source) {
  return _collectBytes(source, (_, result) => result);
}

/// Generalization over [collectBytes] and [collectBytesCancelable].
///
/// Performs all the same operations, but the final result is created
/// by the [result] function, which has access to the stream subscription
/// so it can cancel the operation.
T _collectBytes<T>(Stream<List<int>> source,
    T Function(StreamSubscription<List<int>>, Future<Uint8List>) result) {
  var bytes = BytesBuilder(copy: false);
  var completer = Completer<Uint8List>.sync();
  var subscription =
      source.listen(bytes.add, onError: completer.completeError, onDone: () {
    completer.complete(bytes.takeBytes());
  }, cancelOnError: true);
  return result(subscription, completer.future);
}
```

vs.

```dart
/// Consolidates a stream of [Uint8List] into a single [Uint8List]
Future<Uint8List> consolidateBytes(Stream<Uint8List> stream) async {
  final builder = BytesBuilder(copy: false);

  await for (final chunk in stream) {
    builder.add(chunk);
  }

  return builder.takeBytes();
}
```

we could probably swap it out for the one in dart:async. I don't have the resources to open a PR and test it right now though.